dvc/repo/experiments/queue/remove.py

```diff
@@ -0,0 +1,185 @@
+        for msg, queue_entry in celery_queue._iter_queued():
+            if queue_entry.stash_rev in stash_revs:
+                celery_queue.celery.reject(msg.delivery_tag)
+    finally:
+        celery_queue._remove_revs(stash_revs, celery_queue.stash)
+
+
+def _remove_done_tasks(
+    celery_queue: "LocalCeleryQueue",
+    queue_entries: Iterable[Optional["QueueEntry"]],
+):
+    """Remove done tasks.
+
+    Arguments:
+        queue_entries: An iterable list of done task to remove
+    """
+    # pylint: disable=protected-access
+    from celery.result import AsyncResult
+
+    failed_stash_revs: Dict[str, "ExpStashEntry"] = {}
+    queue_entry_set: Set["QueueEntry"] = set()
+    for entry in queue_entries:
+        if entry:
+            queue_entry_set.add(entry)
+            if entry.stash_rev in celery_queue.failed_stash.stash_revs:
+                failed_stash_revs[
+                    entry.stash_rev
+                ] = celery_queue.failed_stash.stash_revs[entry.stash_rev]
+
+    try:
+        for msg, queue_entry in celery_queue._iter_processed():
+            if queue_entry not in queue_entry_set:
+                continue
+            task_id = msg.headers["id"]
+            result: AsyncResult = AsyncResult(task_id)
+            if result is not None:
+                result.forget()
+            celery_queue.celery.purge(msg.delivery_tag)
+    finally:
+        celery_queue._remove_revs(failed_stash_revs, celery_queue.failed_stash)
+
+
+def _get_names(entries: Iterable[Union["QueueEntry", "QueueDoneResult"]]):
+    names: List[str] = []
+    for entry in entries:
+        if isinstance(entry, QueueDoneResult):
+            if entry.result and entry.result.ref_info:
+                names.append(entry.result.ref_info.name)
+                continue
+            entry = entry.entry
+        name = entry.name
+        name = name or entry.stash_rev[:7]
+        names.append(name)
+    return names
+
+
+def clear(self: "LocalCeleryQueue", **kwargs) -> List[str]:
+    """Remove entries from the queue.
+
+    Arguments:
+        queued: Remove all queued tasks.
+        failed: Remove all failed tasks.
+        success: Remove all success tasks.
+
+    Returns:
+        Revisions which were removed.
+    """
+    queued = kwargs.pop("queued", False)
+    failed = kwargs.get("failed", False)
+    success = kwargs.get("success", False)
+
+    removed = []
+    if queued:
+        queue_entries = list(self.iter_queued())
+        _remove_queued_tasks(self, queue_entries)
+        removed.extend(_get_names(queue_entries))
+    if failed or success:
+        done_tasks: List["QueueDoneResult"] = []
+        if failed:
+            done_tasks.extend(self.iter_failed())
+        if success:
+            done_tasks.extend(self.iter_success())
+        done_entries = [result.entry for result in done_tasks]
+        _remove_done_tasks(self, done_entries)
+        removed.extend(_get_names(done_tasks))
+
+    return removed
+
+
+def remove(
```

Comment:
```suggestion
def celery_remove(
```

Since this is in `queue/remove.py` and is specific to the Celery version of the queue, make sure to name it appropriately so it's obvious this is not used in the base queue or other local queue types